core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -571,6 +806,17 @@ class ReplicaManager(val config: KafkaConfig,
   private def nonOfflinePartitionsIterator: Iterator[Partition] = {
     allPartitions.values.iterator.flatMap {
       case HostedPartition.Online(partition) => Some(partition)
+      case HostedPartition.Fenced(partition) => Some(partition)
+      case HostedPartition.None | HostedPartition.Offline => None
+    }
+  }
+
+  // An iterator over all online partitions. This is a weakly consistent iterator; a partition made offline or fenced
+  // after the iterator has been constructed could still be returned by this iterator.
+  private def onlinePartitionsIterator: Iterator[Partition] = {
+    allPartitions.values.iterator.flatMap {
+      case HostedPartition.Online(partition) => Some(partition)
+      case HostedPartition.Fenced(_) => None
```

Comment:
nit: seems reasonable to consolidate these three cases since this function is only trying to catch online partitions